pkg/join/join.go

```diff
@@ -0,0 +1,554 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package join
```

Comment:
Perhaps we should put this in the `deploy` package and name the entry function `Operator`. So similar to `deploy.Broker`, callers would reference `deploy.Operator`.

Reply:
`deploy.Operator` is another function to deploy operator. May be this can be `deploy.Submariner`?

Reply:
yeah `deploy.Submariner` sounds reasonable. Its responsibility should basically be to deploy the `Submariner` and `ServiceDiscovery` CRs. I think the gateway node labeling should be separate - we can put that code as part of _cmd/subctl/join.go_ at least for now and perhaps move it to its own lib package under _pkg_ later.